src/core/mac/sub_mac.cpp

```diff
@@ -139,6 +139,10 @@ otRadioCaps SubMac::GetCaps(void) const
     caps |= OT_RADIO_CAPS_RX_ON_WHEN_IDLE;
 #endif
 
+#if OPENTHREAD_RADIO
+    caps |= OT_RADIO_CAPS_SLEEP_TO_TX;
+#endif
```

Comment:
This change here LGTM.

We also have other changes in this PR related to `aSkipRcpCompatibilityCheck` ->  `aSkipRcpVersionCheck` and how the `CheckRadioCapabilities()` is called. May be useful to carve this part out as its own saparate PR?

Reply:
SGTM:+1: